### PR TITLE
[NPU] Support npu kernel for assign_value op

### PR DESCRIPTION
**Pull request by betterpig**
### PR types
New features

### PR changes
OPs

### Describe
1. add assign_value_op_npu.cc, it simply register npu kernel, and call compute function in assign_value_op.h to finish job.
2. add code for npu condition in **paddle/fluid/framework/tensor_util.h: TensorFromArray**. The reason is as follows:
When inputs' dtype is bool, there is no corresponding TensorFromVector template function for vector<bool>, so only  TensorFromArray can be used, but it didn't consider npu condition before.

ctest result and op running environment
Pasted Graphic 6

### CODE REVIEW

**Review by zhwesky2010 — DISMISSED** [review]
LGTM

File: python/paddle/fluid/tests/unittests/npu/test_assign_value_op_npu.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+import unittest
+import numpy
+import sys
+sys.path.append("..")
```
Comment (zhwesky2010) [inline python/paddle/fluid/tests/unittests/npu/test_assign_value_op_npu.py:20]: 这个是必须的吗

File: paddle/fluid/operators/assign_value_op_npu.cc [hunk 2]
```diff
@@ -0,0 +1,22 @@
+//   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
```
Comment (qili93) [inline paddle/fluid/operators/assign_value_op_npu.cc]: 日期改一下2021

File: python/paddle/fluid/tests/unittests/npu/test_assign_value_op_npu.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
```
Comment (qili93) [inline python/paddle/fluid/tests/unittests/npu/test_assign_value_op_npu.py]: 时间同样改成2021

File: python/paddle/fluid/tests/unittests/npu/test_assign_value_op_npu.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        self.attrs["int64_values"] = [int(v) for v in self.value.flat]
+
+
+class TestAssignValueNPUOp4(TestAssignValueNPUOp):
+    def init_data(self):
+        self.value = numpy.random.choice(
+            a=[False, True], size=(2, 5)).astype(numpy.bool)
+        self.attrs["bool_values"] = [bool(v) for v in self.value.flat]
+
+
```
Comment (qili93) [inline python/paddle/fluid/tests/unittests/npu/test_assign_value_op_npu.py:76]: 建议参考test_assign_value_op.py把里面4个TestAssignApi相关的单测也加上，相应的修改NPUPlace。

**Review by qili93 — CHANGES_REQUESTED** [review]

File: python/paddle/fluid/tests/unittests/npu/test_assign_value_op_npu.py (hunk 1 above)
Comment (betterpig) [reply python/paddle/fluid/tests/unittests/npu/test_assign_value_op_npu.py:20]: 你好，感谢review。都是必须的。unittests：需要调用unittests.main()执行单测；numpy：用来生成随机数据；最后两行若未加上，会报“module op_test not found"的错误。

File: paddle/fluid/operators/assign_value_op_npu.cc (hunk 2 above)
Comment (betterpig) [reply paddle/fluid/operators/assign_value_op_npu.cc]: 你好，感谢review。已改正。

File: python/paddle/fluid/tests/unittests/npu/test_assign_value_op_npu.py (hunk 3 above)
Comment (betterpig) [reply python/paddle/fluid/tests/unittests/npu/test_assign_value_op_npu.py]: 你好，感谢review。已改正。

File: python/paddle/fluid/tests/unittests/npu/test_assign_value_op_npu.py (hunk 4 above)
Comment (betterpig) [reply python/paddle/fluid/tests/unittests/npu/test_assign_value_op_npu.py:76]: 你好，感谢review。已将相应的TestAssignApi类加上并改成NPUPlace，单测结果如下图。
Pasted Graphic 7

**Review by qili93 — APPROVED** [review]
LGTM

**Pull request merged**